biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java

```diff
@@ -3766,4 +3773,31 @@ private void addDefinedContracts() {
 		contracts.collectContracts(Constants.DEFINE_CONTRACT, definedContracts);
 	}
 
+	private Attrs mergeImportAttrs(Attrs existing, Attrs imported) {
+		if (existing == null) {
+			return (imported == null) ? new Attrs() : new Attrs(imported);
+		}
+		if (imported != null) {
+			if (!imported.containsKey(RESOLUTION_DIRECTIVE)) {
+				// Widen to implied "mandatory" since at least one consumer
+				// imports it as such
+				existing.remove(RESOLUTION_DIRECTIVE);
+			} else if (imported.containsKey(RESOLUTION_DIRECTIVE)
+				&& RESOLUTION_MANDATORY.equals(imported.get(RESOLUTION_DIRECTIVE))) {
```

Comment:
```suggestion
			} else if (RESOLUTION_MANDATORY.equals(imported.get(RESOLUTION_DIRECTIVE))) {
```

Why retest for `imported.containsKey(RESOLUTION_DIRECTIVE)`? We know the key is present as we are in the else side of the if.

Reply:
sure, fixed...